lib/azure/armrest/armrest_collection.rb

```diff
@@ -4,6 +4,32 @@ module Azure
   module Armrest
     class ArmrestCollection < Array
       attr_accessor :continuation_token
+      attr_accessor :response_headers
+
+      # Creates and returns a ArmrestCollection object based on JSON input,
+      # using +klass+ to generate the list elements. In addition, both the
+      # response headers and continuation token are set.
+      #
+      def self.create_from_response(response, klass = nil)
+        json_response = JSON.parse(response)
+        array = new(json_response['value'].map { |hash| klass.new(hash) })
+
+        array.response_headers = response.headers
+        array.continuation_token = parse_skip_token(json_response)
+
+        array
+      end
+
+      alias skip_token continuation_token
+      alias skip_token= continuation_token=
+
+      private
```

Comment:
@djberg96 please fix. `private` does not make class method private. See the rubocop messages. Those should be addressed.